File: go/search/dateRangeCriterion.go
package search

// NewDateRangeCriterionCriterion is a convenience function to construct a wrapped new DateRangeCriterionCriterion.
func NewDateRangeCriterionCriterion(
	field string,
	start *RangeValue,
	end *RangeValue,
) *Criterion {
	return &Criterion{
		Criterion: &Criterion_DateRangeCriterion{
			DateRangeCriterion: &DateRangeCriterion{
				Field: field,
				Start: start,
				End:   end,
			},
		},
	}
}

File: ts/src/search/dateRangeCriterion.ts
import { Criterion, CriterionSchema } from "./criterion_pb";
import { RangeValue, DateRangeCriterionSchema } from "./dateRangeCriterion_pb";
import { create } from "@bufbuild/protobuf";

export function newDateRangeCriterion(
  field: string,
  start?: RangeValue,
  end?: RangeValue,
): Criterion {
  return create(
    CriterionSchema,
    {
      criterion: {
        case: "dateRangeCriterion",
        value: create(DateRangeCriterionSchema, { field, start, end }),
      },
    },
  );
}
